### PR TITLE
Issue #4342: bumped version of pmd-mvn-plugin to 3.8 and pmd to 5.7.0

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue: #4342 
 
pmd-maven-plugin version was bumped to 3.8
dependencies on pmd 5.7.0 were added to pmd-maven-plugin
all violations were suppressed for now

### CODE REVIEW

**Comment by Nimfadora** [comment]
@rnveach, @romani I cannot understand why it has crashed. The error has no connection to my changes. It is becoming annoying.

**Comment by rnveach** [comment]
> VM terminated without properly saying goodbye

@Nimfadora This is an issue with travis. We are not sure what is causing it and have been discussing it more.
As long as normal tests pass and the first 5 nodes, than I am fine for now.

**Comment by Nimfadora** [comment]
@rnveach I think I have found bug, will try to recommit

File: pom.xml [hunk 1]
```diff
@@ -358,6 +359,28 @@
               <excludeRoot>target/generated-sources/antlr/com/puppycrawl/tools/checkstyle/grammars/javadoc</excludeRoot>
             </excludeRoots>
           </configuration>
+          <dependencies>
+            <dependency>
+              <groupId>net.sourceforge.pmd</groupId>
+              <artifactId>pmd-core</artifactId>
```
Comment (romani) [inline pom.xml]: remove reference to this dependency

File: pom.xml [hunk 2]
```diff
@@ -358,6 +359,28 @@
...
+              <version>${pmd.version}</version>
+            </dependency>
+            <dependency>
+              <groupId>net.sourceforge.pmd</groupId>
+              <artifactId>pmd-java</artifactId>
+              <version>${pmd.version}</version>
+            </dependency>
+            <dependency>
+              <groupId>net.sourceforge.pmd</groupId>
+              <artifactId>pmd-javascript</artifactId>
```
Comment (romani) [inline pom.xml]: remove reference to this dependency

File: pom.xml [hunk 3]
```diff
@@ -358,6 +359,28 @@
...
+              <version>${pmd.version}</version>
+            </dependency>
+            <dependency>
+              <groupId>net.sourceforge.pmd</groupId>
+              <artifactId>pmd-jsp</artifactId>
```
Comment (romani) [inline pom.xml]: remove reference to this dependency

File: pom.xml (hunk 1 above)
Comment (Nimfadora) [inline pom.xml]: Removed

File: pom.xml (hunk 2 above)
Comment (Nimfadora) [inline pom.xml]: Removed

File: pom.xml (hunk 3 above)
Comment (Nimfadora) [inline pom.xml]: Removed

**Pull request merged**